### ISSUE TITLE
BaremetalPI fails on Julia 1.10.1 and above

### ISSUE THREAD

**Issue by JakeZw** [ronisbr/BaremetalPi.jl#7]
Something changed in Julia 1.10.1 that causes BaremetalPI to error.  The discussion regarding the details of the problem is on [discourse](https://discourse.julialang.org/t/dev-gpiomem-error-for-baremetalpi-mmap-mmap-regression/112611), where there are links for more detail.

**Comment by ronisbr** [comment]
Hi @JakeZw !

Yes, there is something different in the new version of Julia. However, I am trying to have access again to a RaspberryPi to debug the issue. Maybe I will be able to test it next week.

**Comment by JakeZw** [comment]
I tried it again with Julia 1.10.5 and 1.11.0-rc3 and both have the same issue.  Then I though I would set the compat for Mmap to "= 1.10.0" but that failed.  I guess Mmap is part of the standardlib so cannot be set independently.

Finally I tried to see if I could use the Mmap from Julia 1.10.0 and make it a submodule of BareMetalPi, but also failed in that.  I am using the PI 4 with 8 GB RAM and Bookworm OS.  Any suggestions?
